web_console/utils/job_builder.js

```diff
@@ -1,91 +1,104 @@
 let server_config;
 try {
-    server_config = require('../server.config');
+  server_config = require('../server.config');
 } catch (err) {
-    server_config = require('../constants').DEFAULT_SERVER_CONFIG;
+  server_config = require('../constants').DEFAULT_SERVER_CONFIG;
 }
 
-function roleConfig(config, public_env = {}, public_volume = {}) {
-    let role = {
-        "pair": config.pair,
-        "replicas": config.replicas,
-        "template": {
-            "spec": {
-                "restartPolicy": "Never",
-                "volumes": [],
-                "containers": {
-                    "env": [
-                        { "name": "POD_IP", "value": { "valueFrom": { "fieldRef": { "fieldPath": "status.podIP" } } } },
-                        { "name": "POD_NAME", "value": { "valueFrom": { "fieldRef": { "fieldPath": "metadata.name" } } } }],
-                    "image": config.image,
-                    "imagePullPolicy": "IfNotPresent",
-                    "volumeMounts": [],
-                    "name": "tensorflow",
-                    "ports": config.ports,
-                    "resources": { "limits": { "cpu": config.cpu, "memory": config.memory }, "requests": { "cpu": config.cpu, "memory": config.memory } },
-                    "command": config.command,
-                    "args": config.args
-                },
-            }
-        }
+function mergeJson(a, b) {
+  a = Object.assign({}, a);
+  for (key in b) {
+    if (key in a && Array.isArray(a[key]) && Array.isArray(b[key])) {
+      a[key] = a[key].concat(b[key]);
+    } else if (key in a && a[key].constructor == Object &&
```

Comment:
`null` value would throw error here, so we always check Object type in this way: giving variable `foo`, check that with `foo && typeof foo === 'object'`